### DISCUSSION TITLE
Feed/fuse multiple FAN tacho signals into one MBFAN tacho output

### DISCUSSION THREAD

**Discussion by siodor** [tjko/fanpico#58] [Ideas]
Hello fanpico Community

First, thanks to @tjko for making this great project! :smiley: 

Would it be possible to introduce a way to monitor multiple FAN tacho signals from one MBFAN tacho output?

I have the following situation:
- Mainboard with only 1x PWM fan header
- 3 fans (CPU, PSU, HBA) on the fanpico
   - CPU fan (FAN1) follows signal from MBFAN1
   - PSU fan (FAN2) minimum 50% PWM duty cycle, if MBFAN1 asks for more than 50% then ramp up accordingly
   - HBA fan (FAN3) minimum 50% PWM duty cycle, if MBFAN1 asks for more than 50% then ramp up accordingly
   - If any FANx goes below requested RPM then set MBFAN1 tacho output to zero to notify mainboard and shutdown PC

Could this be implemented?

**Comment by tjko** [comment]
I would think this should be relatively easy to implement in the firmware.

Existing command: [CONF:MBFAN:SOURCE](https://github.com/tjko/fanpico/blob/main/commands.md#configurembfanxsource)

currently support two types of "sources" for the fan signals:  FAN and FIXED


I was thinking, perhaps could add couple new types, something like:
- MIN - returns lowest fan speed of 2 or more fans
- MAX - returns fastest fan speed of 2 or more fans
- AVG - returns average fan speed of 2 or more fans


So something like "CONF:MBFAN1:SOURCE MIN,1,2,3" could be used to configure MBFAN1 header to output whatever is the slowest fan speed observed across fans 1 to 3....

**Comment by tjko** [comment]
@siodor, initial support for "calculated" sources for MBFAN tacho outputs is now checked in (https://github.com/tjko/fanpico/pull/59).

What you were looking for should now work if you use "min" fan source across your three fans for the MBFAN1 output.
Something like (assuming fans are FAN1, FAN2, FAN3):
```
conf:mbfan1:source min,1,2,3
```

While existing [CONF:MBFANx:RPMMAP](https://github.com/tjko/fanpico/blob/main/commands.md#configurembfanxrpmmap) command should allow making MBFAN1 output to output "0 RPM" signal if fan speed is below desired limit...


If you want to test, firmware images are available (for a short time) at: https://github.com/tjko/fanpico/actions/runs/7148029602
(or you can always compile firmware yourself...)

**Reply by siodor** [reply]
Hello @tjko ,

Wow, thanks for the quick implementation.

I'll give it a go today.

**Reply by siodor** [reply]
Hello @tjko 

I managed to test this over the weekend and it works as suggested with
`conf:mbfan1:source min,1,2,3`
like a charm

My testing was very simple. I unplugged the fans one by one and the status LED of the PC immediately went red each time.

**Reply by tjko** [reply]
Thanks for testing. I only tested with two fans, glad to hear it seems to work fine with more fans as well.
